pysages/ssages/ffs.py

```diff
@@ -0,0 +1,181 @@
+            #how pysages takes snapshots from system?
+            system.restore_snapshot(snapshot)
+            has_reachedA=False
+            while not has_reachedA:
+                run(1)
+                time_count+=timestep
+                snap = system.take_snapshot()
+                ξ, Jξ=cv(rs, indices(ids))
+                Crossed=CrossedtoA(ξ,grid)
+                if Crossed:
+                    success+=1;
+                    has_reachedA=True;
+                    if len(window0_snaps)<=Num_window0:
+                        window0_snaps.append(snap)
+        Phi_a=float(success)/(time_count)
+        return Phi_a,window0_snaps
+#Function to run each window and calculate transition probabilities
+    def running_window(cv,grid,step,old_snapshots,system,rs,ids):
+        success=0
+        win_A=grid[0]
+        win_value=grid[step]
+        new_snapshots=[]
+        for i in range(0,len(old_snapshots)):
+            snapshot=old_snapshots[i]
+            #sages snapshots can be used to restore the simulation?
+            system.restore_snapshot(snapshot)
+            select=0
+            while select==0:
+                run(1)
+                snap=system.take_snapshot()
+                ξ, Jξ=cv(rs, indices(ids))
+                select=CrossingInterface(ξ,grid,step)
+                if select==-1:
+                    break
+                if select==1:
+                    new_snapshots.append(snapshot)
+                    success+=1
+                    break
+        if success==0:
+            return print('Error in window'+' '+str(win_value)+'\n')
+        if len(new_snapshots)>0:
+            prob_local=float(success)/float(len(old_snapshots))
+        return prob_local,new_snapshots
+    def update(snapshot, cv, grid, Nmax_replicas, N0_steps,sampling_time,system,run):
+        Check=good_input(system,grid,cv)
+        if not Check:
+            print('Bad initial configuration'+'\n')
+            exit()
+
+#Initial runs to sample from basin A
+        Ini_snapshots=Basin_sampling(system,limit,sampling_time,grid,rs,ids)
+#Calculate initial flow
+        Phi_a,Ini_0=Initial_flow(limit,dt,grid,Ini_snapshots,system,rs,ids)
+        write_to_file(Phi_a)
+        write_trajectory(0)
+        INI=Ini_0*100
+        Hist=np.zeros(len(windows))
+        Hist[0]=Phi_a
+#Calculate conditional probability for each window
+        for k in range(1,len(windows)):
+            if k==1:
+                old_snaps=INI
+            prob, w1_snapshots=running_window(Nw_steps,grid,k,old_snaps,system,rs,ids)
+            write_to_file(prob)
+            Hist[k]=prob
+            old_snaps=increase_snaps(w1_snapshots,INI)
+            print('size of snapshots= '+str(len(old_snaps))+'\n')
+        K_t=np.prod(Hist)
+        write_to_file('#Flux_Constant')
+        write_to_file(K_t)
+        return FFSState(Phi_A,Previous_Window,Current_Window,Prob_window)
+    return snapshot, initialize, generalize(update)
```

Comment:
We can do this for now to prevent jit compilation issues and jit compile only a subset of the code here manually.

```suggestion
    return snapshot, initialize, generalize(update, jit_compile = False)
```